api/handler.js

```diff
@@ -0,0 +1,130 @@
+const java = require('./src/java');
+const ping = require('./src/ping');
+
+const tracking = require('./src/utils/tracking');
+const cache = require('./src//utils/cache');
+const log = require('./src/utils/log');
+
+const logPrefix = log.prefix;
+
+const supportedTypes = ['ping', 'go', 'java', ...registries.supported];
+
+const mapper = async (item) => {
+  let result;
+
+  if (registries.supported.includes(item.type)) {
+    result = await registries.resolve(item.type, item.target);
+  } else if (item.type === 'go') {
+    result = await go(item.target);
+  } else if (item.type === 'java') {
+    result = await java(item.target);
+  } else if (item.type === 'ping') {
+    result = await ping(item.target);
+  } else {
+    return;
+  }
+
+  return {
+    ...item,
+    result,
+  };
+};
+
+function cleanPayload(payload) {
+  // Remove duplicates
+  // Remove invalid items which does not follow format {type:'foo', target: 'bar'}
+  // Filter out types which are not supported
+  return uniqWith(payload, isEqual).filter(
+    item => item
+      && item.target
+      && item.target.length
+      && supportedTypes.includes(item.type),
+  );
+}
+
+async function requestHandler(payload) {
+  return pMap(payload, mapper, { concurrency: 5 });
+}
+
+function errorHandler(error, res) {
+  log(error);
+  res.statusCode = 500;
+  res.end('Internal server error');
+}
+
+tracking.init();
+
+module.exports = async (req, res) => {
+  if (req.method === 'POST') {
+    const timingTotalStart = Date.now();
+
+    const body = await json(req);
+
+    await cache.auth();
+
+    let result;
+    let timingTotalEnd;
+    let completed = false;
+    const payload = cleanPayload(body);
+
+    try {
+      result = await requestHandler(payload);
+      completed = true;
+    } catch (error) {
+      return errorHandler(error, res);
+    } finally {
+      timingTotalEnd = Date.now();
+
+      log('Request completed', completed);
+      log('Instance Name', logPrefix);
+      log('Redis Status', cache.getRedisStatus());
+      log('Simple Cache Size', cache.simpleCacheSize());
+      log('Result Length', (result && result.length) || 0);
+      log('Duration', timingTotalEnd - timingTotalStart);
+
+      const meta = {
+        event: '2meta',
+        properties: {
+          Region: process.env.NOW_REGION || 'unknown',
+          'Request completed': completed,
+          'Instance Name': logPrefix,
+          ...cache.getRedisStatus(),
+          'Simple Cache Size': cache.simpleCacheSize(),
+          'Result Length': (result && result.length) || 0,
+          Duration: timingTotalEnd - timingTotalStart,
+        },
+      };
+
+      const bulkData = [
+        meta,
+        ...result.map(item => ({
+          event: '2packages',
+          properties: {
+            Type: item.type,
+            Target: item.target,
+            Result: item.result,
+          },
+        })),
+      ];
+
+      console.time('Mixpanel');
+      await tracking.track(bulkData);
```

Comment:
By analysing 1300 requests, this calls adds an average 77ms for each request which isn't ideal, but acceptable in our case. Once we have better tooling for visibility we can remove it.